test/test_nn.py

```diff
@@ -2646,6 +2646,19 @@ def test_weight_norm(self):
             m = torch.nn.utils.weight_norm(m)
             m = torch.nn.utils.weight_norm(m)
 
+    def test_weight_norm_parameterlistdict(self):
```

Comment:
Nit: this test more accurately verifies that parameter list and parameter dict throw warnings when a user tries to add attributes to them.